Source/WebKit/WebProcess/Extensions/Cocoa/WebExtensionContextProxyCocoa.mm

```diff
@@ -127,6 +128,36 @@
     return result;
 }
 
+bool WebExtensionContextProxy::hasPermission(const String& permission) const
+{
+    WallTime currentTime = WallTime::now();
+
+    // If the next expiration date hasn't passed yet, there is nothing to remove.
+    if (m_nextGrantedPermissionsExpirationDate != WallTime::nan() && m_nextGrantedPermissionsExpirationDate > currentTime)
+        goto finish;
```

Comment:
why not just `return m_grantedPermissions.contains(permission);`? i'm also not familiar with `goto` but i imagine this just executes the code in `finish`

Reply:
That's what it does. It avoid duplicating code, and prevents one place from being different than the other.